xbmc/utils/ColorUtils.cpp

```diff
@@ -9,34 +9,165 @@
 #include "ColorUtils.h"
 
 #include "Color.h"
+#include "StringUtils.h"
+#include "utils/log.h"
+#include "utils/XBMCTinyXML.h"
 
+#include <algorithm>
 #include <math.h>
+#include <stdio.h>
 
-UTILS::Color ColorUtils::ChangeOpacity(const UTILS::Color color, const float opacity)
+using namespace UTILS;
+
+
+namespace
+{
+
+void GetHSLValues(ColorEx& colorEx)
 {
-  int newAlpha = ceil( ((color >> 24) & 0xff) * opacity);
+  double r = (colorEx.colorARGB & 0x00FF0000) >> 16;
+  double g = (colorEx.colorARGB & 0x0000FF00) >> 8;
+  double b = (colorEx.colorARGB & 0x000000FF);
+  r /= 255;
+  g /= 255;
+  b /= 255;
+  const double& maxVal = std::max<double>({r, g, b});
+  const double& minVal = std::min<double>({r, g, b});
+  double h = 0;
+  double s = 0;
+  double l = (minVal + maxVal) / 2;
+  double d = maxVal - minVal;
+
+  if (d == 0)
+  {
+    h = s = 0; // achromatic
+  }
+  else
+  {
+    s = l > 0.5 ? d / (2 - maxVal - minVal) : d / (maxVal + minVal);
+    if (maxVal == r)
+    {
+      h = (g - b) / d + (g < b ? 6 : 0);
+    }
+    else if (maxVal == g)
+    {
+      h = (b - r) / d + 2;
+    }
+    else if (maxVal == b)
+    {
+      h = (r - g) / d + 4;
+    }
+    h /= 6;
+  }
+
+  colorEx.hue = h;
+  colorEx.saturation = s;
+  colorEx.lightness = l;
+}
+
+} // namespace
```

Comment:
i and many other defs close anon namespaces with `} // unnamed namespace`. just a personal thing at the end